### PR TITLE
Unescape escaped straight apostrophes

### PR DESCRIPTION
**Pull request by mathjazz**
As mentioned in #3020, we need to unescape straight apostrophes in existing
- entities,
- translations,
- TM entries.

That provides consistency between existing and future translations containing straight apostrophes and makes it easier to edit existing translations.

This PR adds a data migration that unescapes straight apostrophes. Reverse migration is noop, because we already have unescaped straight apostrophes in the DB (e.g. generated by MT).

### CODE REVIEW

**Comment by mathjazz** [comment]
Pushed to stage.

File: pontoon/base/migrations/0050_strings_xml_apostrophes.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    )
+    translations.update(string=Replace("string", Value("\\'"), Value("'")))
+
+    TranslationMemoryEntry = apps.get_model("base", "TranslationMemoryEntry")
+    tm_entries = TranslationMemoryEntry.objects.filter(
+        entity__resource__format="xml",
+    )
+    source_tm_entries = tm_entries.filter(source__contains="\\'")
+    source_tm_entries.update(source=Replace("source", Value("\\'"), Value("'")))
+    target_tm_entries = tm_entries.filter(source__contains="\\'")
```
Comment (eemeli) [inline pontoon/base/migrations/0050_strings_xml_apostrophes.py:29]:
```suggestion
    target_tm_entries = tm_entries.filter(target__contains="\\'")
```

**Review by eemeli — APPROVED** [review]
Fix the typo; otherwise it looks good.

File: pontoon/base/migrations/0050_strings_xml_apostrophes.py (hunk 1 above)
Comment (mathjazz) [reply pontoon/base/migrations/0050_strings_xml_apostrophes.py:29]: Ugh! Thanks!

**Pull request merged**